src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -126,12 +127,31 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 );
                 String defaultsFilePath = DefaultUseCases.getDefaultsFileByUseCaseName(useCase);
                 useCaseDefaultsMap = ParseUtils.parseJsonFileToStringToStringMap("/" + defaultsFilePath);
-
-                if (request.hasContent()) {
+                List<String> requiredParams = DefaultUseCases.getRequiredParamsByUseCaseName(useCase);
+
+                if (request.hasContent() == false) {
+                    if (requiredParams.size() != 0) {
```

Comment:
Combine these with an &&.  Also:
- don't check `== false`.  Just do `!request.hasContent()`.
- `size != 0` is better as `!requiredParams.isEmpty()`.

Reply:
answered above